src/Sarif.Viewer.VisualStudio/ErrorList/SarifSuppressedResultsTableDataSource.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.Sarif.Viewer.ErrorList
+{
+    // A "stub" data source whose sole purpose is to cause the Supression State column
+    // to be displayed. It never actually adds entries to the Error List.
+    //
+    // For an explanation of why these this is necessary, see the comment near the top
+    // of SarifResultTableEntry.cs
+    internal class SarifSuppressedResultsTableDataSource : SarifTableDataSourceBase
+    {
+        private static SarifSuppressedResultsTableDataSource _instance;
+
+        private SarifSuppressedResultsTableDataSource()
+        {
+            if (!SarifViewerPackage.IsUnitTesting)
+            {
+#pragma warning disable VSTHRD108 // Assert thread affinity unconditionally
+                ThreadHelper.ThrowIfNotOnUIThread();
+#pragma warning restore VSTHRD108 // Assert thread affinity unconditionally
+
+                this.Initialize(SarifResultTableEntry.SuppressedResultColumns);
+            }
+        }
+
+        public static SarifSuppressedResultsTableDataSource Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new SarifSuppressedResultsTableDataSource();
+                }
+
+                return _instance;
+            }
+        }
```

Comment:
We could change to:
```csharp
return _instance ?? (_instance = new SarifSuppressedResultsTableDataSource());
```
or
```csharp
public static SarifSuppressedResultsTableDataSource Instance => _instance ?? (_instance = new SarifSuppressedResultsTableDataSource());
```

Reply:
Yes, nice.

---
In reply to: [538269677](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538269677) [](ancestors = 538269677)

Reply:
NIT: C# 8.0 one-liner:

return _instance ??= new SarifSuppressedResultsTableDataSource();

---
In reply to: [538737288](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538737288) [](ancestors = 538737288,538269677)

Reply:
Indeed, my favorite C# 8 feature.

---
In reply to: [538759807](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538759807) [](ancestors = 538759807,538737288,538269677)